spec/omniauth/strategies/saml_spec.rb

```diff
@@ -125,22 +125,35 @@ def post_xml(xml=:example_response, opts = {})
     context "when fingerprint is empty and there's a fingerprint validator" do
       before :each do
         saml_options.delete(:idp_cert_fingerprint)
-        saml_options[:idp_cert_fingerprint_validator] = lambda { |fingerprint| "C1:59:74:2B:E8:0C:6C:A9:41:0F:6E:83:F6:D1:52:25:45:58:89:FB" }
-        post_xml
+        saml_options[:idp_cert_fingerprint_validator] = fingerprint_validator
       end
 
-      it "should set the uid to the nameID in the SAML response" do
-        expect(auth_hash['uid']).to eq '_1f6fcf6be5e13b08b1e3610e7ff59f205fbd814f23'
+      let(:fingerprint_validator) { lambda { |_| "C1:59:74:2B:E8:0C:6C:A9:41:0F:6E:83:F6:D1:52:25:45:58:89:FB" } }
+
+      context "when the fingerprint validator returns a truthy value" do
+        before { post_xml }
+
+        it "should set the uid to the nameID in the SAML response" do
+          expect(auth_hash['uid']).to eq '_1f6fcf6be5e13b08b1e3610e7ff59f205fbd814f23'
+        end
+
+        it "should set the raw info to all attributes" do
+          expect(auth_hash['extra']['raw_info'].all.to_hash).to eq(
+            'first_name'   => ['Rajiv'],
+            'last_name'    => ['Manglani'],
+            'email'        => ['user@example.com'],
+            'company_name' => ['Example Company'],
+            'fingerprint'  => 'C1:59:74:2B:E8:0C:6C:A9:41:0F:6E:83:F6:D1:52:25:45:58:89:FB'
+          )
+        end
       end
 
-      it "should set the raw info to all attributes" do
-        expect(auth_hash['extra']['raw_info'].all.to_hash).to eq(
-          'first_name'   => ['Rajiv'],
-          'last_name'    => ['Manglani'],
-          'email'        => ['user@example.com'],
-          'company_name' => ['Example Company'],
-          'fingerprint'  => 'C1:59:74:2B:E8:0C:6C:A9:41:0F:6E:83:F6:D1:52:25:45:58:89:FB'
-        )
+      context "when the fingerprint validator returns false" do
+        let(:fingerprint_validator) { lambda { |_| false } }
+
+        before { post_xml }
+
+        it { should fail_with(:invalid_ticket) }
```

Comment:
I just noticed we're using a mix of expect and should syntax. Can you make this `is_expected.to`?